lua/luasnip/init.lua

```diff
@@ -202,12 +202,20 @@ local function snip_expand(snippet, opts)
 		end
 	end
 
-	session.current_nodes[vim.api.nvim_get_current_buf()] =
-		util.no_region_check_wrap(
-			snip.jump_into,
-			snip,
-			1
-		)
+	if not opts.skip_snippet then
```

Comment:
```suggestion
	if snip:is_interactive() then
```
this seems to work for me, any drawbacks? wdyt?

Reply:
we can probably make more robust by adding a fallback condition (also) 
```lua
  if snip:is_interactive() or vim.api.nvim_get_mode() == "i" then
```
basically jump if you're already in insert mode just in case, since it's highly unlikely that you'd want to expand a snippet in insert mode and not have the cursor as well.

Reply:
IMO this is pretty specific and I wouldn't really want to include it (we'd also have to disable it for triggered expansion where you'd want your cursor to be at the position of the `$0` every time)
What do you think of accepting a callback responsible for jumping into the snippet instead? Then you could get as fancy as you like :D

Reply:
hmm, something is up with`vim.api.nvim_get_mode()` but `vim.fn.mode()` works fine

Reply:
Getting the mode is a bit finnicky :/
I think we also had some problems with using api for it and use `vim.fn.mode` when we need the current mode.

Reply:
> What do you think of accepting a callback responsible for jumping into the snippet instead?

so my original approach was trying to check `jumpable()` before attempting to call `jump_into`, but I wasn't able to figure out how to do that, since `jumpable()` is accessed by the node and not the snippet, and I'm not sure which _node_ it would be in this case 😅

Reply:
You can get the nodes inside a snippets either as `snippet.nodes`, for the nodes in the order they appear in the buffer, or `snippet.insert_nodes`, where the keys are the position of the node (ie. `insert_nodes[1]` ->`i(1)`, `insert_nodes[0]` -> `i(0)`, `insert_nodes[2]` -> `c(2, {...})`)

Not sure if `jumpable` really is what you want here, `is_interactive` is the right approach I think.